cmd/tikv-ctl/src/main.rs

```diff
@@ -89,6 +89,10 @@ fn handle_engine_error(err: EngineError) -> ! {
             );
         }
     }
+
+    // force async logger to flush by dropping its guard.
+    *tikv_util::logger::ASYNC_LOGGER_GUARD.lock().unwrap() = None;
+
     process::exit(-1);
```

Comment:
`process::exit(-1)` is called in elsewhere as well, please extract a function.

Reply:
sorry, I try to rebase my pull request on master, and then I can not reproduce the issue at this moment.
I will dig into the source code and see the reasons behind this and I will keep you updated with the progress.

Reply:
my bad, this issue can still be reproduced by executing many and many times or by adding more error! print sentences before process::exit(-1).